interpreter/src/main/scala/pl/touk/nussknacker/engine/definition/TypeInfos.scala

```diff
@@ -21,112 +21,103 @@ object TypeInfos {
 
   implicit val methodInfoEncoder: Encoder[MethodInfo] = Encoder[SerializableMethodInfo].contramap(_.serializable)
 
-  object MethodInfo {
-    private val arrayClass = classOf[Array[Object]]
-
-    def apply(parameters: List[Parameter],
-              refClazz: TypingResult,
-              name: String,
-              description: Option[String],
-              varArgs: Boolean): StaticMethodInfo = (varArgs, parameters) match {
-      case (true, noVarArgParameters :+ Parameter(paramName, TypedClass(`arrayClass`, varArgType :: Nil))) =>
-        VarArgsMethodInfo(noVarArgParameters, Parameter(paramName, varArgType), refClazz, name, description)
-      case (true, _ :+ Parameter(_, TypedClass(`arrayClass`, _))) =>
-        throw new AssertionError("Array must have one type parameter")
-      case (true, _ :+ Parameter(_, _)) =>
-        throw new AssertionError("VarArg must have type of array")
-      case (true, Nil) =>
-        throw new AssertionError("Method with varArgs must have at least one parameter")
-      case (false, _) =>
-        NoVarArgsMethodInfo(parameters, refClazz, name, description)
-    }
-  }
-
   sealed trait MethodInfo {
     def computeResultType(arguments: List[TypingResult]): ValidatedNel[ExpressionParseError, TypingResult]
 
-    def staticParameters: List[Parameter]
+    def staticParameters: List[Parameter] =
+      staticNoVarArgParameters ::: staticVarArgParameter.map{ case Parameter(name, refClazz) =>
+        Parameter(name, Typed.genericTypeClass(classOf[Array[Object]], List(refClazz)))
+      }.toList
+
+    def staticParametersWithSimpleVarArg: List[Parameter] = staticNoVarArgParameters ::: staticVarArgParameter.toList
+
+    def staticNoVarArgParameters: List[Parameter]
+
+    def staticVarArgParameter: Option[Parameter]
 
     def staticResult: TypingResult
 
     def description: Option[String]
 
-    def varArgs: Boolean
+    def varArgs: Boolean = staticVarArgParameter.isDefined
```

Comment:
can u do "non interface" methods final? it will be easier to analyze interface when I'll see less methods that can be overridden